src/oled_display_node.cpp

```diff
@@ -292,32 +400,44 @@ int dispOled_initCtx(std::string devName, dispCtx_t *dispCtx, int dispType, uint
  *
  * @param               devName         Name of the I2C output device
  * @param               dispCtx         Context for display that this command will populate
- * @param               dispType        Type of display. DISPLAY_TYPE_SSD1306 or DISPLAY_TYPE_SH1106
+ * @param               displayType     Type of display. DISPLAY_TYPE_SSD1306 or DISPLAY_TYPE_SH1106
  * @param               i2cAddr         7-bit I2C bus address
  *
- * @return              Returns 0 for ok or -1 for IO error
+ * @return              Returns 0 for ok or negative for IO error
  */
-int dispOled_init(std::string devName, dispCtx_t *dispCtx, int dispType, uint8_t i2cAddr)
+int dispOled_init(std::string devName, dispCtx_t *dispCtx, int displayType, uint8_t i2cAddr)
 {
     int retCode = 0;
+    int dispType = displayType;
+
+    // If this is called with NONE we try to autodetect the display
+    if (dispType == DISPLAY_TYPE_NONE) {
+        // Auto-detect display. Detects if display present and type of OLED display
+        retCode = dispOled_detectDisplayType(devName, i2cAddr, &dispType);
+        if (retCode != 0) {
+            return retCode;
+        }
+    }
 
     dispOled_initCtx(devName, dispCtx, dispType, i2cAddr);
 
     //Send all the commands to fully initialize the device.
     switch (dispCtx->dispType) {
     case DISPLAY_TYPE_SSD1306:
         // We treat the 1st byte sort of like a 'register' but it is really a command stream mode to the chip
-                retCode = dispOled_writeBytes(dispCtx, &ssd1306_init_bytes[0], SSD1306_INIT_BYTE_COUNT);
+        printf("%s:  Setup for SSD1306 controller on the OLED display\n", THIS_NODE_NAME);
```

Comment:
Here and on a couple of other places you use printf instead of ROS_INFO, is there a reason for it?

Reply:
This is an oversight.   I will change this.  thank you

Reply:
Await the push that resolves this prior to any comments on this printf issue is fine with me but your call.

Reply:
All the printf statements are removed so it's ready to approve